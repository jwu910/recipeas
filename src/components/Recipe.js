import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button, Collapse, Well } from 'react-bootstrap';

class Recipe extends Component {
  constructor(props) {
    super(props);

    this.state = {};
    this.toggleRecipe = this.toggleRecipe.bind(this);
  }
  toggleRecipe() {
    this.setState({ open: !this.state.open });
  }
  render() {
    return (
      <div className='recipe-container'>
        <h4 className="recipe-title" onClick={this.toggleRecipe}>{this.props.title}</h4>
        <Collapse in={this.state.open}>
          <div> {/* this div exists for smooth collapse animation */}
            <Well>
              <ul>{this.props.ingredients}</ul>
              <Button className="edit-recipe" onClick={() => {
                this.props.fillEditRecipeModal(this.props.title);
              }}>Edit</Button>
              <Button className="delete-recipe" bsStyle="danger" onClick={() => this.props.removeRecipe(this.props.title)}>Delete</Button>
            </Well>
          </div>
        </Collapse>
      </div>
    );
  }
}

Recipe.propTypes = {
  title: PropTypes.string.isRequired,
  ingredients: PropTypes.array.isRequired,
  showEditRecipe: PropTypes.func.isRequired,
  removeRecipe: PropTypes.func.isRequired,
  fillEditRecipeModal: PropTypes.func.isRequired
}

export default Recipe;
