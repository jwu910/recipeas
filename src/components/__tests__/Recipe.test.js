import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Recipe from '../Recipe';

const title = 'BLT';

const ingredientsList = 'Bread, Bacon, Lettuce, Tomato'.split(',').map(ingredient => {
  return (
    <li className="list-group-item" key={title + ingredient}>
      { ingredient }
    </li>
  );
});

const props = {
  title: title,
  ingredients: ingredientsList,
  removeRecipe: jest.fn(),
  showEditRecipe: jest.fn(),
  fillEditRecipeModal: jest.fn()
};

const setup = () => {
  const enzymeWrapper = shallow(<Recipe {...props} />);

  return {
    props,
    enzymeWrapper
  };
}

describe('Recipe', () => {
  test('renders without errors', () => {
    const component = renderer.create(<Recipe { ...props }/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render self with correct HTML', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper.find('h4').text()).toBe(title);

    expect(enzymeWrapper.find('ul').contains(ingredientsList)).toEqual(true);

  });
});
