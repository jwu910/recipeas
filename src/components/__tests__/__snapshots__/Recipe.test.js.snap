// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe renders without errors 1`] = `
<div
  className="recipe-container"
>
  <h4
    className="recipe-title"
    onClick={[Function]}
  >
    BLT
  </h4>
  <div
    aria-expanded={null}
    className="collapse"
  >
     
    <div
      className="well"
    >
      <ul>
        <li
          className="list-group-item"
        >
          Bread
        </li>
        <li
          className="list-group-item"
        >
           Bacon
        </li>
        <li
          className="list-group-item"
        >
           Lettuce
        </li>
        <li
          className="list-group-item"
        >
           Tomato
        </li>
      </ul>
      <button
        className="edit-recipe btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Edit
      </button>
      <button
        className="delete-recipe btn btn-danger"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Delete
      </button>
    </div>
  </div>
</div>
`;
